docs/DeveloperGuide.md

```diff
@@ -1,34 +1,610 @@
+8. Click `OK` to accept the default settings.
+9. Wait for the importing process to finish.
+10. Locate run the 'Duke.java' file, right click and select `Run Duke.main()`.
+11. Navigate to the `test\java` folder and right click `seedu.duke` and select `Run Tests in 'seedu.duke'`.
+If the setup is successful, you should see this after step 10. All tests should also pass.
+````
+_________________________________
+Welcome to scheduler--;!
+What can I do for you?
+_________________________________
+File Created: Personal
+File Created: Zoom
+File Created: Timetable
+File Created: Goal
+The file has successfully been loaded!
+_________________________________
+You have no events today!
+_________________________________
+````
+## Design
+### Architecture
 
+![Architecture Diagram](./diagrams/Architecture.jpg)
 
-## Product scope
-### Target user profile
+The ***Architecture Diagram*** given above explains the high-level design of App. Given below is a quick overview of
+each component.
 
-{Describe the target user profile}
+`Duke` class contains the main method. It is responsible for,
 
-### Value proposition
+- At app launch: Initializes the components in the correct sequence, and connects them up with each other.  
+- At shut down: Stores the user data from memory into the hard disk.
 
-{Describe the value proposition: what problem does it solve?}
+The rest of the App consists of five components.
 
-## User Stories
+- `Ui`: The Ui of the App.
+- `Parser`: Parses the input to determine which command to execute.
+- `Command`: The command executer.
+- `UserData`: Holds the data of the App in the memory.
+- `Storage`: Reads data from, and writes data to, the hard disk.
+
+#### Ui Component
+The 'Ui' component is in charge of handling input from users and system output.
+
+#### Command Component
+
+![Diagram for commmand](./diagrams/Command.jpg)
+
+API: [Command.java](https://github.com/AY2021S1-CS2113T-T12-4/tp/blob/master/src/main/java/seedu/duke/command/Command.java) 
+1. `Parser` class to parse the user command 
+1.  This results in a `Command` object executed by the `Parser`
+1.  The command execution can affect the `Model`
+
+It listens for commands made in the Duke Class and sends the input to the parser class.
+It is also responsible for printing messages from commands and exception messages. 
+
+#### Parser Component
+![Parser diagram](./diagrams/Parser_diagram.jpg)
+
+The `Parser` object uses its `parse` method to parse inputs passed to it from `Ui`. This method returns a `Command` object to be executed by `Duke`.
+
+#### UserData Component
+![UserData diagram](./diagrams/UserData_diagram.jpg)
+
+The `UserData`
+- stores `EventList` objects that store events of different categories (i.e. Personal, Timetable, Zoom).
+- stores a `Goal` object that represents the current user goal.
+- does not depend on the other components.
+
+#### Storage Component
+![Diagram for storage class](./diagrams/classDiagramStorage.jpg)
```

Comment:
The diagram implies that storage knows about UI but UI does not know about storage. Should change to double-headed arrow if both are aware of each other, or reverse the direction of the arrow if UI knows about Storage.
Also, consider replacing the 4 classnames connected to Storage with the actual classnames. For the class attribute name, consider using the standard notation of name:type = default-value e.g. name:String = personal.txt. Unless the 4 classes are not actually classes, in which case they should not be shown on a class diagram as it is non-standard notation.

Consider having some multiplicities in general.

Storage to StorageParser might be better represented by a solid line, since it seems that Storage cannot function fully without the StorageParser i.e. the dependency is not temporary.